src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -46,7 +46,7 @@ public abstract partial class Frame : FrameContext, IFrameControl
 
         private readonly object _onStartingSync = new Object();
         private readonly object _onCompletedSync = new Object();
-        protected readonly FrameRequestHeaders _requestHeaders = new FrameRequestHeaders();
+        internal readonly FrameRequestHeaders _requestHeaders = new FrameRequestHeaders();
```

Comment:
Might be better to make TakeMessageHeaders non-static. I think it only changes one test in FrameTests that now needs to create a frame, but that isn't to hard (see FrameFacts... I'm not sure why that is a separate class).